examples/smc.jl

```diff
@@ -4,32 +4,45 @@ using Distributions
 using Plots
 using StatsFuns
 
-# Particle Filter implementation
-struct Particle{T} # Here we just need a tree
+# Particle Filter 
+struct Particle{T<:AbstractStateSpaceModel,V} <: AbstractParticle{T}
     parent::Union{Particle,Nothing}
-    state::T
+    state::V
 end
 
-Particle(state::T) where {T} = Particle(nothing, state)
-Particle() = Particle(nothing, nothing)
-Base.show(io::IO, p::Particle) = print(io, "Particle($(p.state))")
+function Particle(parent, state, model::T) where {T<:AbstractStateSpaceModel}
+    return Particle{T,particleof(model)}(parent, state)
+end
+
+function Particle(model::T) where {T<:AbstractStateSpaceModel}
+    N = dimension(model)
+    V = particleof(model)
+    state = N == 1 ? zero(V) : zeros(V, N)
+    return Particle{T,V}(nothing, state)
+end
+
+Base.show(io::IO, p::Particle{T,V}) where {T,V} = print(io, "Particle{$T, $V}($(p.state))")
```

Comment:
```suggestion
Base.show(io::IO, p::Particle{T,V}) where {T,V} = print(io, "Particle{", T, V, "}(", p.state, ")")
```

This suggests their is a constructor of this form - but it seems there isn't?